web/src/beta/lib/reearth-ui/components/NumberInput/index.tsx

```
@@ -47,22 +47,26 @@
   const handleChange = useCallback(
     (e: ChangeEvent<HTMLInputElement>) => {
       const currentValue = e.currentTarget.value;
-      if (/^-?\d*\.?\d*%?$/.test(currentValue)) {
-        const numericValue = Number(currentValue);
+
+      if (/^-?\d*\.?\d*(e-?\d*)?$/.test(currentValue.toLowerCase())) {
         let validatedValue = currentValue;
 
-        if (currentValue !== "") {
+        if (currentValue !== "" && !isNaN(Number(currentValue))) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)